client/e2e/integration/user-list.spec.ts

```diff
@@ -0,0 +1,135 @@
+import { UserListPage } from '../support/user-list.po';
+
+const page = new UserListPage();
+
+describe('User list', () => {
+
+  beforeEach(() => {
+    page.navigateTo();
+  });
+
+  it('Should have the correct title', () => {
+    page.getUserTitle().should('have.text', 'Users');
+  });
+
+  it('Should type something in the name filter and check that it returned correct elements', () => {
+    // Filter for user 'Lynn Ferguson'
+    cy.get('#user-name-input').type('Lynn Ferguson');
+
+    // All of the user cards should have the name we are filtering by
+    page.getUserCards().each(e => {
+      cy.wrap(e).find('.user-card-name').should('have.text', 'Lynn Ferguson');
```

Comment:
Directly looking for `.user-card-name` makes our tests dependent (in many places) on that class. Does that suggest that we really want to have a PO or action here that encapsulates that name so it's not repeated everywhere?

Reply:
I suppose we could have a PO function that returns `cy.wrap(e).find('.user-card-name')`